test/interval.jl

```diff
@@ -887,4 +887,119 @@
             end
         end
     end
+
+    @testset "floor" begin
+        @test_throws UndefKeywordError floor(Interval(0.0, 1.0))
+
+        @test floor(Interval(0.0, 1.0), on=LeftEndpoint) == Interval(0.0, 1.0)
+        @test floor(Interval(0.5, 1.0), on=LeftEndpoint) == Interval(0.0, 0.5)
+        @test floor(Interval(0.0, 1.5), on=LeftEndpoint) == Interval(0.0, 1.5)
+        @test floor(Interval(0.5, 1.5), on=LeftEndpoint) == Interval(0.0, 1.0)
+
+        @test floor(Interval(0.0, 1.0), on=RightEndpoint) == Interval(0.0, 1.0)
+        @test floor(Interval(0.5, 1.0), on=RightEndpoint) == Interval(0.5, 1.0)
+        @test floor(Interval(0.0, 1.5), on=RightEndpoint) == Interval(-0.5, 1.0)
+        @test floor(Interval(0.5, 1.5), on=RightEndpoint) == Interval(0.0, 1.0)
```

Comment:
We should have tests for attempting to use `AnchoredEndpoint` with an `Interval`